be/btree.c

```diff
@@ -718,12 +720,16 @@ static void be_btree_shift_key_vals(struct m0_be_bnode *dest,
 	{
 		dest->bt_kv_arr[i + key_dest_offset] =
 					src->bt_kv_arr[i + key_src_offset];
-		*dest->bt_ik[i + key_dest_offset].inlkey =
-					*((char *)src->bt_kv_arr[i + key_src_offset].btree_key);
-		dest->allocated[i + key_dest_offset] = 1;
+		// *dest->bt_ik[i + key_dest_offset].inlkey =
+		// 			*((char *)src->bt_kv_arr[i + key_src_offset].btree_key);
+		
 		if(ktype != M0_BBT_CAS_CTG)
 		{
-			dest->bt_kv_arr[i + key_dest_offset].btree_key = &dest->bt_ik[i + key_dest_offset].inlkey;
+			int tindex = find_next_avaliable_index(dest->allocated);
+			memcpy(dest->bt_ik[tindex + key_dest_offset].inlkey,src->bt_kv_arr[i + key_src_offset].btree_key,64);
```

Comment:
should be only tindex as the array offset and not tindex + key_dest_offset.
The next 2 lines also have problems.